addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/CcoHandler.java

```diff
@@ -0,0 +1,185 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lutron.handler;
+
+import static org.openhab.binding.lutron.LutronBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.Locale;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.lutron.internal.protocol.LutronCommandType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler responsible for communicating with Lutron contact closure outputs (CCOs).
+ * e.g. VCRX CCO outputs and CCO RF module outputs
+ *
+ * @author Bob Adair - Initial contribution
+ *
+ *         Note: For a RA2 Pulsed CCO, querying the output state with ?OUTPUT,<id>,1 is meaningless and will always
+ *         return 100 (on). Also, the main repeater will not report ~OUTPUT commands for a pulsed CCO regardless of
+ *         the #MONITORING setting. So this binding supports sending pulses ONLY.
+ */
+
+public class CcoHandler extends LutronHandler {
+    private static final Integer ACTION_PULSE = 6;
+    private static final Integer ACTION_STATE = 1;
+
+    private final Logger logger = LoggerFactory.getLogger(CcoHandler.class);
+
+    private int integrationId;
+    private double defaultPulse = 0.5; // default pulse length (seconds)
+
+    protected enum CcoOutputType {
+        PULSED,
+        MAINTAINED
+    }
+
+    protected CcoOutputType outputType;
+
+    public CcoHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public int getIntegrationId() {
+        return this.integrationId;
+    }
+
+    @Override
+    public void initialize() {
+        Number id = (Number) getThing().getConfiguration().get(INTEGRATION_ID);
+
+        if (id == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No integrationId");
+            return;
+        }
+        this.integrationId = id.intValue();
+
+        // Determine output type from configuration if not pre-defined by subclass
+        if (this.outputType == null) {
+            String oType = (String) getThing().getConfiguration().get(OUTPUT_TYPE);
+
+            if (oType == null || oType == OUTPUT_TYPE_PULSED) {
+                logger.info("Setting CCO type Pulsed for device {}.", integrationId);
+                this.outputType = CcoOutputType.PULSED;
+            } else if (oType == OUTPUT_TYPE_MAINTAINED) {
+                logger.info("Setting CCO type Maintained for device {}.", integrationId);
```

Comment:
Reduce to debug.

Reply:
I'd argue that it is good to have "set to pulsed" or "set to maintained" recorded in the log for CCO devices on every startup, since setting the wrong one could have a serious impact on some devices.  I've changed the other info level log message to debug.